commons/src/main/java/com/powsybl/commons/datasource/TarArchiveDataSource.java

```diff
@@ -0,0 +1,330 @@
+import org.apache.commons.compress.compressors.xz.XZCompressorOutputStream;
+import org.apache.commons.compress.compressors.zstandard.ZstdCompressorInputStream;
+import org.apache.commons.compress.compressors.zstandard.ZstdCompressorOutputStream;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * @author Nicolas Rol {@literal <nicolas.rol at rte-france.com>}
+ */
+public class TarArchiveDataSource extends AbstractArchiveDataSource {
+
+    public TarArchiveDataSource(Path directory, String tarFileName, String baseName, String mainExtension, CompressionFormat compressionFormat, DataSourceObserver observer) {
+        super(directory, tarFileName, baseName, mainExtension, compressionFormat, ArchiveFormat.TAR, observer);
+    }
+
+    public TarArchiveDataSource(Path directory, String tarFileName, String baseName, String mainExtension, CompressionFormat compressionFormat) {
+        this(directory, tarFileName, baseName, mainExtension, compressionFormat, null);
+    }
+
+    public TarArchiveDataSource(Path directory, String baseName, String mainExtension, CompressionFormat compressionFormat, DataSourceObserver observer) {
+        this(directory,
+            baseName + (mainExtension == null || mainExtension.isEmpty() ? "" : "." + mainExtension) + ".tar" + (compressionFormat == null ? "" : "." + compressionFormat.getExtension()),
+            baseName, mainExtension, compressionFormat, observer);
+    }
+
+    public TarArchiveDataSource(Path directory, String baseName, String mainExtension, CompressionFormat compressionFormat) {
+        this(directory,
+            baseName + (mainExtension == null || mainExtension.isEmpty() ? "" : "." + mainExtension) + ".tar" + (compressionFormat == null ? "" : "." + compressionFormat.getExtension()),
+            baseName, mainExtension, compressionFormat, null);
+    }
+
+    public TarArchiveDataSource(Path directory, String baseName, CompressionFormat compressionFormat, DataSourceObserver observer) {
+        this(directory,
+            baseName + ".tar" + (compressionFormat == null ? "" : "." + compressionFormat.getExtension()),
+            baseName, null, compressionFormat, observer);
+    }
+
+    public TarArchiveDataSource(Path directory, String baseName, CompressionFormat compressionFormat) {
+        this(directory,
+            baseName + ".tar" + (compressionFormat == null ? "" : "." + compressionFormat.getExtension()),
+            baseName, null, compressionFormat, null);
+    }
+
+    public TarArchiveDataSource(Path directory, String baseName) {
+        this(directory,
+            baseName + ".tar",
+            baseName, null, null, null);
+    }
+
+    public TarArchiveDataSource(Path tarFile) {
```

Comment:
use private constructor with a FileInformation argument to avoid calling 3 times ?